src/reporter/health.py

```diff
@@ -0,0 +1,89 @@
+import os
+
+
+def check_crate():
+    """
+    crateDB is the default backend of QuantumLeap, so it is required by
+    default.
+    """
+    from translators.crate import CrateTranslatorInstance
+    with CrateTranslatorInstance() as trans:
+        crate_health = trans.get_health()
+        return crate_health
+
+
+def check_geocache():
+    """
+    Geocache is relevant only when geocoding usage is enabled.
+    """
+    use_geocoding = os.environ.get('USE_GEOCODING', False)
+    if not use_geocoding:
+        return {'status': 'pass'}
+
+    from geocoding.geocache import GeoCodingCache
+    host = os.environ.get('REDIS_HOST', None)
+    port = os.environ.get('REDIS_PORT', 6379)
+    gc = GeoCodingCache(host, port)
+    health = gc.get_health()
+    return health
+
+
+def check_geocoder():
+    """
+    Geocoder is relevant only when geocoding usage is enabled.
+    """
+    use_geocoding = os.environ.get('USE_GEOCODING', False)
+    if not use_geocoding:
+        return {'status': 'pass'}
+
+    from geocoding import geocoding
+    health = geocoding.get_health()
+    return health
+
+
+def _get_http_code(res):
+    if res['status'] == 'pass':
+        code = 200
+    elif res['status'] == 'warn':
+        code = 207
```

Comment:
never seen this used, what's the case with ELK api? that maybe a good reference

Reply:
good idea, let me see...

Reply:
Elasticserach returns 200 even for a cluster with "yellow" status, so it may be better to stick with 200. Besides, 207 might require something else to properly implement multistatus.

Reply:
then let's use 200, I think 207 is not much adopted.

Reply:
ok!